### PR TITLE
concept entry sql rollback #5168

### LINKED ISSUE #5168: [Concept Entry] SQL ROLLBACK
### Reason/inspiration (optional)

We would like a new entry on the `ROLLBACK` concept in SQL. The entry should go in a new file under `docs/content/sql/concepts/commands/terms/rollback/rollback.md`.    

The entry should include:

- An introduction to the concept   

- A `Syntax` section that provides the syntax for the concept (if applicable)   

- An `Example` section that provides an example demonstrating the concept in use (if applicable)   

Please refer to the [concept entry template](https://github.com/Codecademy/docs/blob/main/documentation/concept-entry-template.md), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Concept Entrty for Existing Topic

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [ ] Issue is assigned

### PR DESCRIPTION
**Pull request by judygitsthejobdone**
### Description

New entry on the ROLLBACK concept in SQL under docs/content/sql/concepts/commands/rollback/rollback.md
Includes introduction, syntax, and examples for ROLLBACK and ROLLBACK TO SAVEPOINT.

This is my first contribution, done as part of the Learn Git & GitHub course on codecademy.

### Issue Solved
Closes #5168 

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

Note: yarn test is failing 1 test, displaying the below. I have double-checked the file structure and it is as requested, happy to oblige any requested changes to resolve.

>Codecademy Docs Content › adheres to content file structure                                                                                            
>                                                                                                                                                         
>    expect(received).toBe(expected) // Object.is equality
>
>    Expected: ""
>    Received: "C:\\Projects\\Codecademy\\Codecademy_docs\\content\\sql\\concepts\\commands\\rollback\\rollback.md - expected a directory but got a file"

### CODE REVIEW

**Review by SaviDahegaonkar — CHANGES_REQUESTED** [review]
Hey Author,
I had a look at your file, I have suggested a few changes so make them asap.

**Review by SaviDahegaonkar — CHANGES_REQUESTED** [review]
Hey Author,
I had a look at your file, I have suggested a few changes so make them asap.

File: content/sql/concepts/commands/rollback/rollback.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+| --- | --- | --- |
+| 1 | 11/16/2001 | Jolly Old Chap |
+| 2 | 1/12/2000 | Jolly Young Chap |
+
+```sql
+--Fix Young Chap's birthday to be younger than Old Chap
+UPDATE table_1
+    SET dob = "1/21/2024";
+SELECT * FROM table_1;
+```
```
[the remark this answers is not in the document]
Comment (judygitsthejobdone) [reply content/sql/concepts/commands/rollback/rollback.md:69]: The omission of the WHERE clause is intentional in the example. It forces us to demonstrate the ROLLBACK of the transaction exemplified in lines 77-92

File: content/sql/concepts/commands/rollback/rollback.md [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+## Syntax
+
+```sql
+ROLLBACK [WORK] [TO SAVEPOINT savepoint_name];
+
+| CLAUSE                      | Description                                                                                                                                                                                 |
+| --------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| TO SAVEPOINT savepoint_name | (optional) Resets database to an established savepoint `savepoint_name` within the current transaction. These savepoint names can be created using the command `SAVEPOINT savepoint_name;`. |
+
```
Comment (SaviDahegaonkar) [inline content/sql/concepts/commands/rollback/rollback.md:34]:
```suggestion
```sql
ROLLBACK [WORK] [TO SAVEPOINT savepoint_name];

| CLAUSE                      | Description                                                                                                                                                                                 |
| --------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| TO SAVEPOINT savepoint_name | (optional) Resets database to an established savepoint `savepoint_name` within the current transaction. These savepoint names can be created using the command `SAVEPOINT savepoint_name;`. |

```
The syntax is written inside the pseudo blocks (```.....```).

File: content/sql/concepts/commands/rollback/rollback.md [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+### Rollback Current Transaction
+
+The command can be entered alone to end and undo any changes in the current transaction.
+
+```sql
+ROLLBACK;
+--Or, for compatability with some SQL standards:
```
Comment (SaviDahegaonkar) [inline content/sql/concepts/commands/rollback/rollback.md:41]:
```suggestion
--Or, for compatibility with some SQL standards:
```

**Review by SaviDahegaonkar — CHANGES_REQUESTED** [review]
Heu @judygitsthejobdone ,
I have suggested you some changes please make them asap.


Thanks,
Savi

File: content/sql/concepts/commands/rollback/rollback.md (hunk 2 above)
Comment (judygitsthejobdone) [reply content/sql/concepts/commands/rollback/rollback.md:34]: I'm not following the change. Do you mean that line 30 should include triple backticks to end the block? Or that line 28 should say "pseudo" in place of "sql"

File: content/sql/concepts/commands/rollback/rollback.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+ROLLBACK TO SAVEPOINT second;
+SELECT name FROM table_1 WHERE id=3;
+
+--gives "Jolly Medium Chap"
+
+--But what is medium age anyway? Just go back to your save with just two chaps.
+ROLLBACK TO SAVEPOINT first;
+SELECT * FROM table_1;
+
+````
```
Comment (SaviDahegaonkar) [inline content/sql/concepts/commands/rollback/rollback.md:129]:
```suggestion
```
```

**Review by Sriparno08 — APPROVED** [review]
Looks good, @judygitsthejobdone!

**Pull request merged**